packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts

```diff
@@ -393,9 +397,35 @@ export class AzureCommunicationChatAdapter implements ChatAdapter {
     this.emitter.off(event, listener);
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private executeWithThreadAccess(callback: () => any): any {
+    if (!this.hasAcceccToThread) {
+      console.error('Chat Adapter does not have access to the thread.');
+    } else {
+      return callback();
+    }
+  }
+
+  private checkIsUserJoinedThread(): void {
+    const check = (): void => {
+      const threads = this.chatClient.listChatThreads({ maxPageSize: 100 }).byPage().next();
+      threads.then((threads) => {
+        if (threads.value.filter((thread: ChatThreadItem) => thread.id === this.chatThreadClient.threadId).length > 0) {
+          this.hasAcceccToThread = true;
+        } else {
+          setTimeout(check, 500);
+        }
+      });
+    };
+
+    check();
+  }
+
   private async asyncTeeErrorToEventEmitter<T>(f: () => Promise<T>): Promise<T> {
     try {
-      return await f();
+      return await this.executeWithThreadAccess(() => {
```

Comment:
This function is more for event emitter - it does not align with function name to add this function here, maybe we can do this in a separated function?

Reply:
For sure we can wrap the function for another time - but maybe extra what we have and make another logic for this